apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs

```diff
@@ -128,20 +123,52 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             }
 
             // If we have format params and its more than just the original message add them.
+            Struct paramStruct = null;
             if (state is IEnumerable<KeyValuePair<string, object>> formatParams &&
                 !(formatParams.Count() == 1 && formatParams.Single().Key.Equals("{OriginalFormat}")))
             {
-                var paramStruct = new Struct();
+                paramStruct = new Struct();
                 foreach (var pair in formatParams)
                 {
                     // Consider adding formatting support for values that are IFormattable.
-                    paramStruct.Fields.Add(pair.Key, Value.ForString(pair.Value?.ToString() ?? ""));
+                    paramStruct.Fields[pair.Key] = Value.ForString(pair.Value?.ToString() ?? "");
                 }
+            }
+
+            var currentLogScope = GoogleLoggerScope.Current;
+            if (currentLogScope != null)
+            {
+                jsonStruct.Fields.Add("scope", Value.ForString(currentLogScope.ToString()));
+            }
 
-                if (paramStruct.Fields.Count > 0)
+            while (currentLogScope != null)
+            {
+                // Determine if the state of the scope are format params
+                if (currentLogScope.State is IEnumerable<KeyValuePair<string, object>> scopeFormatParams)
                 {
-                    jsonStruct.Fields.Add("format_parameters", Value.ForStruct(paramStruct));
+                    if (paramStruct == null)
+                    {
+                        paramStruct = new Struct();
+                    }
+
+                    foreach (var pair in scopeFormatParams)
+                    {
+                        if (pair.Key == "{OriginalFormat}")
+                        {
+                            // Skip the {OriginalFormat} parameter for scopes
+                            continue;
+                        }
+
+                        paramStruct.Fields[pair.Key] = Value.ForString(pair.Value?.ToString() ?? "");
+                    }
                 }
+
+                currentLogScope = currentLogScope.Parent;
+            }
+
+            if (paramStruct?.Fields.Count > 0)
```

Comment:
You are asking here for Count > 0 so you can always initialize paramStruct on line 126